codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoWriter.java

```diff
@@ -290,24 +294,60 @@ boolean writePackageShapeDocs(Shape shape) {
      * @return Returns true if docs were written.
      */
     boolean writeMemberDocs(Model model, MemberShape member) {
-        return member.getMemberTrait(model, DocumentationTrait.class)
+        boolean hasDocs;
+
+        hasDocs = member.getMemberTrait(model, DocumentationTrait.class)
                 .map(DocumentationTrait::getValue)
                 .map(docs -> {
                     writeDocs(docs);
-                    member.getMemberTrait(model, MediaTypeTrait.class)
-                            .map(StringTrait::getValue)
-                            .ifPresent(mediaType -> writeDocs(
-                                    "\n\nThis value conforms to the media type: " + mediaType));
-
-                    member.getMemberTrait(model, RequiredTrait.class)
-                            .ifPresent((value) -> {
-                                if (docs.length() != 0) {
-                                    writeDocs("");
-                                }
-                                writeDocs("This member is required.");
-                            });
                     return true;
                 }).orElse(false);
+
+        Optional<String> stringOptional = member.getMemberTrait(model, MediaTypeTrait.class)
+                .map(StringTrait::getValue);
+        if (stringOptional.isPresent()) {
+            if (hasDocs) {
+                writeDocs("");
+            }
+            writeDocs("This value conforms to the media type: " + stringOptional.get());
+            hasDocs = true;
+        }
+
+        GoUsageIndex usageIndex = GoUsageIndex.of(model);
+        if (usageIndex.isUsedForOutput(member)) {
+            if (member.getMemberTrait(model,
+                    HttpPrefixHeadersTrait.class).isPresent()) {
+                if (hasDocs) {
+                    writeDocs("");
+                }
+                writeDocs("Map keys will be normalized to lower-case.");
+                hasDocs = true;
+            }
+        }
+
+        if (member.getMemberTrait(model, RequiredTrait.class).isPresent()) {
+            if (hasDocs) {
+                writeDocs("");
+            }
+            writeDocs("This member is required.");
+            hasDocs = true;
+        }
```

Comment:
minor nit, if there are no docs, this would be 

```
// value conforms to media type...
// this field is required
```

Whereas if there were docs it would be
```
// something really cool to say
//
// value conforms to media type...
//
// this field is required
```

Reply:
`hasDocs` gets set to true when of these sections is first hit if its not already true. So these would be spaced as expected.

Reply:
Thanks missed that being set! You're right